core/src/main/java/jenkins/slaves/DeprecatedAgentProtocolMonitor.java

```diff
@@ -0,0 +1,112 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.slaves;
+
+import hudson.Extension;
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+import hudson.model.AdministrativeMonitor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.CheckForNull;
+import jenkins.AgentProtocol;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.Symbol;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+
+/**
+ * Monitors enabled protocols and warns if a protocol is deprecated. 
+ * 
+ * @author Oleg Nenashev
+ * @since TODO
+ */
+@Extension
+@Symbol("remotingProtocolVersions")
+@Restricted(NoExternalUse.class)
+public class DeprecatedAgentProtocolMonitor extends AdministrativeMonitor {
+    
+    private static final Logger LOGGER = Logger.getLogger(DeprecatedAgentProtocolMonitor.class.getName());
+
+    public DeprecatedAgentProtocolMonitor() {
+        super(AgentProtocol.class.getName() + "-deprecated");
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Deprecated Remoting protocols";
+    }
+
+    @Override
+    public boolean isActivated() {
+        final Set<String> agentProtocols = Jenkins.getInstance().getAgentProtocols();
+        for (String name : agentProtocols) {
+            AgentProtocol pr = AgentProtocol.of(name);
+            if (pr != null && pr.isDeprecated()) {
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    @Restricted(NoExternalUse.class)
+    public String getDeprecatedProtocols() {
+        String res = getDeprecatedProtocolsString();
+        return res != null ? res : "N/A";
+    }
+    
+    @CheckForNull
+    public static String getDeprecatedProtocolsString() {
+        final List<String> deprecatedProtocols = new ArrayList<>();
+        final Set<String> agentProtocols = Jenkins.getInstance().getAgentProtocols();
+        for (String name : agentProtocols) {
+            AgentProtocol pr = AgentProtocol.of(name);
+            if (pr != null && pr.isDeprecated()) {
+                deprecatedProtocols.add(name);
+            }
+        }
+        if (deprecatedProtocols.isEmpty()) {
+            return null;
+        }
+        return StringUtils.join(deprecatedProtocols, ',');
+    }
+    
+    @Initializer(after = InitMilestone.PLUGINS_STARTED)
+    @Restricted(NoExternalUse.class)
+    public static void initializerCheck() {
+        String protocols = getDeprecatedProtocolsString();
+        if(protocols != null) {
+            LOGGER.log(Level.WARNING, "This Jenkins instance uses deprecated Remoting protocols: {0}"
```

Comment:
I would like to keep this log as well. It is important if somebody has remote logging system configured